internal/fs/vss_windows.go

```diff
@@ -0,0 +1,819 @@
+type VssObjectType uint
+
+// VssObjectType constant values necessary for using VSS api.
+const (
+	VSS_OBJECT_UNKNOWN VssObjectType = iota
+	VSS_OBJECT_NONE
+	VSS_OBJECT_SNAPSHOT_SET
+	VSS_OBJECT_SNAPSHOT
+	VSS_OBJECT_PROVIDER
+	VSS_OBJECT_TYPE_COUNT
+)
+
+// UUID_IVSS defines to GUID of IVssBackupComponents.
+var UUID_IVSS = &ole.GUID{
+	0x665c1d5f, 0xc218, 0x414d,
+	[8]byte{0xa0, 0x5d, 0x7f, 0xef, 0x5f, 0x9d, 0x5c, 0x86},
+}
+
+// IVssBackupComponents VSS api interface.
+type IVssBackupComponents struct {
+	ole.IUnknown
+}
+
+// IVssBackupComponentsVTable is the vtable for IVssBackupComponents.
+type IVssBackupComponentsVTable struct {
+	ole.IUnknownVtbl
+	getWriterComponentsCount      uintptr
+	getWriterComponents           uintptr
+	initializeForBackup           uintptr
+	setBackupState                uintptr
+	initializeForRestore          uintptr
+	setRestoreState               uintptr
+	gatherWriterMetadata          uintptr
+	getWriterMetadataCount        uintptr
+	getWriterMetadata             uintptr
+	freeWriterMetadata            uintptr
+	addComponent                  uintptr
+	prepareForBackup              uintptr
+	abortBackup                   uintptr
+	gatherWriterStatus            uintptr
+	getWriterStatusCount          uintptr
+	freeWriterStatus              uintptr
+	getWriterStatus               uintptr
+	setBackupSucceeded            uintptr
+	setBackupOptions              uintptr
+	setSelectedForRestore         uintptr
+	setRestoreOptions             uintptr
+	setAdditionalRestores         uintptr
+	setPreviousBackupStamp        uintptr
+	saveAsXML                     uintptr
+	backupComplete                uintptr
+	addAlternativeLocationMapping uintptr
+	addRestoreSubcomponent        uintptr
+	setFileRestoreStatus          uintptr
+	addNewTarget                  uintptr
+	setRangesFilePath             uintptr
+	preRestore                    uintptr
+	postRestore                   uintptr
+	setContext                    uintptr
+	startSnapshotSet              uintptr
+	addToSnapshotSet              uintptr
+	doSnapshotSet                 uintptr
+	deleteSnapshots               uintptr
+	importSnapshots               uintptr
+	breakSnapshotSet              uintptr
+	getSnapshotProperties         uintptr
+	query                         uintptr
+	isVolumeSupported             uintptr
+	disableWriterClasses          uintptr
+	enableWriterClasses           uintptr
+	disableWriterInstances        uintptr
+	exposeSnapshot                uintptr
+	revertToSnapshot              uintptr
+	queryRevertStatus             uintptr
+}
+
+// getVTable returns the vtable for IVssBackupComponents.
+func (vss *IVssBackupComponents) getVTable() *IVssBackupComponentsVTable {
+	return (*IVssBackupComponentsVTable)(unsafe.Pointer(vss.RawVTable))
+}
+
+// InitializeForBackup calls the eqivalent VSS api.
+func (vss *IVssBackupComponents) InitializeForBackup() HRESULT {
+	result, _, _ := syscall.Syscall(
+		vss.getVTable().initializeForBackup, 2,
```

Comment:
Is `result,` a typo?

Reply:
No, why?

Reply:
I just haven't seen/noticed it before. If it's proper then it's just my lacking Go skills, no worries :)